src/block/utils/SignText.php

```diff
@@ -45,7 +46,7 @@ class SignText{
 	 * @throws \InvalidArgumentException if invalid keys (out of bounds or string) are found in the array
 	 * @throws \InvalidArgumentException if any line is not valid UTF-8 or contains a newline
 	 */
-	public function __construct(?array $lines = null){
+	public function __construct(?array $lines = null, private ?Color $baseColor = null, private bool $glowingText = false){
```

Comment:
You can't declare the colour as a promoted property. The `null` will leak into the property definition and upset PHPStan.

For the sake of consistency it's probably best to avoid promoted properties entirely here